indico/modules/categories/controllers/display.py

```diff
@@ -187,6 +189,36 @@ def _process(self):
                             total_count=total_count, flash=False)
 
 
+class RHCategoryManagedEventSearch(RHCategoryBase):
+    """Search managed events in a category."""
+
+    def _check_access(self):
+        if not session.user:
+            raise Forbidden
+        RHCategoryBase._check_access(self)
+
+    @use_kwargs({'q': LowercaseString(required=True, validate=not_empty)}, location='query')
+    def _process(self, q):
+        from indico.modules.events.series.schemas import EventDetailsForSeriesManagementSchema
+        query = (
+            Event.query.with_parent(self.category)
+            .filter(
+                Event.title_matches(q), Event.series_id.is_(None), ~Event.is_deleted
+            )
+            .options(load_only('id', 'title', 'start_dt', 'end_dt', 'category_chain'))
+        )
+        # Prefer favorite events
+        query = query.order_by(Event.favorite_of.any(favorite_event_table.c.user_id == session.user.id).desc())
+        # Prefer exact matches and matches at the beginning, then order by event title
+        query = query.order_by(
+            (db.func.lower(Event.title) == q).desc(),
+            db.func.lower(Event.title).startswith(q).desc(),
+            db.func.lower(Event.title),
+        )
+        result = [EventDetailsForSeriesManagementSchema().dump(e) for e in query if e.can_manage(session.user)]
```

Comment:
I think you can use the `get_n_matching` util here - that way you are guaranteed to get enough events if enough exist, and avoid querying and access-checking more than necessary

and at the end you can simply use `EventDetailsForSeriesManagementSchema(many=True).jsonify(events)` to dump the list